client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -11,10 +16,150 @@ import Main from 'calypso/components/main';
+import type { Purchase } from 'calypso/lib/purchases/types';
+import type { TranslateResult } from 'i18n-calypso';
+import FormInputCheckbox from 'calypso/components/forms/form-checkbox';
+import FormLabel from 'calypso/components/forms/form-label';
 
-export default function PaymentMethodAdd(): ReactElement {
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+function onPaymentSelectComplete( {
+	successCallback,
+	translate,
+	showSuccessMessage,
+	purchase,
+}: {
+	successCallback: () => void;
+	translate: ReturnType< typeof useTranslate >;
+	showSuccessMessage: ( message: string | TranslateResult ) => void;
+	purchase?: Purchase | undefined;
+} ) {
+	if ( purchase ) {
+		showSuccessMessage( translate( 'Your payment method has been set.' ) );
+	} else {
+		showSuccessMessage( translate( 'Your payment method has been added successfully.' ) );
+	}
+	successCallback();
+}
+
+function CurrentPaymentMethodNotAvailableNotice( {
+	purchase,
+}: {
+	purchase: Purchase;
+} ): JSX.Element | null {
 	const translate = useTranslate();
+	const moment = useLocalizedMoment();
+	const storedPaymentAgreements = useSelector( getStoredPaymentAgreements );
+	const noticeProps: Record< string, boolean | string | number | TranslateResult > = {
+		showDismiss: false,
+	};
+
+	if ( purchase.payment.creditCard && creditCardHasAlreadyExpired( purchase ) ) {
+		noticeProps.text = translate(
+			'Your %(cardType)s ending in %(cardNumber)d expired %(cardExpiry)s.',
+			{
+				args: {
+					cardType: purchase.payment.creditCard.type.toUpperCase(),
+					cardNumber: parseInt( purchase.payment.creditCard.number, 10 ),
+					cardExpiry: moment( purchase.payment.creditCard.expiryDate, 'MM/YY' ).format(
+						'MMMM YYYY'
+					),
+				},
+			}
+		);
+		return <Notice { ...noticeProps } />;
+	}
+
+	if ( getPaymentMethodIdFromPayment( purchase.payment ) === 'paypal-existing' ) {
+		const storedPaymentAgreement = storedPaymentAgreements.find(
+			( agreement ) => agreement.stored_details_id === purchase.payment.storedDetailsId
+		);
+		if ( storedPaymentAgreement?.email ) {
+			noticeProps.text = translate(
+				'This purchase is currently billed to your PayPal account (%(emailAddress)s).',
+				{
+					args: {
+						emailAddress: storedPaymentAgreement.email,
+					},
+				}
+			);
+			return <Notice { ...noticeProps } />;
+		}
+
+		noticeProps.text = translate( 'This purchase is currently billed to your PayPal account.' );
+		return <Notice { ...noticeProps } />;
+	}
+
+	return null;
+}
+
+function PaymentMethodAdd(): ReactElement {
+	const purchase = undefined;
+	const reduxDispatch = useDispatch();
+	const translate = useTranslate();
+	const { isStripeLoading, stripeLoadingError, stripeConfiguration, stripe } = useStripe();
+
+	const stripeMethod = useCreateStoredCreditCard( {
+		isStripeLoading,
+		stripeLoadingError,
+		stripeConfiguration,
+		stripe,
+		activePayButtonText: translate( 'Save payment method' ) as string,
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **13** times:
`translate( 'Saved payment method' )` *ES Score: 8*

Reply:
@IanRamosC Not really a big thing but we could "speed up the translation and consistency" by just naming it `Save` since the "payment method" part should be understandable in the context of what is happening 😄